backend/src/models/reports/Tab_10_rpt_CA_MultiYrStats_ConsultingServicesbyPortfolio.js

```diff
@@ -1,16 +1,336 @@
+            allocated_deliverable_total, 
+            CAST(0 AS MONEY)
+          ) AS allocated_amount 
+        FROM 
+          (
+            q3 
+            INNER JOIN contract c ON q3.contract_id = c.id
+          ) 
+          INNER JOIN fiscal_year fy ON c.fiscal = fy.id 
+        UNION 
+        SELECT 
+          fiscal_year_id, 
+          fiscal_year, 
+          co_number, 
+          portfolio_abbrev, 
+          resource_type, 
+          resource_count, 
+          allocated_contract_amount 
+        FROM 
+          q4`)
+      )
+      .with(
+        "q1",
+        knex.raw(`
+        SELECT 
+          q2.fiscal_year_id, 
+          q2.fy, 
+          q2.resource_type, 
+          SUM(
+            CASE WHEN q2.portfolio_abbrev = 'OSS' THEN ROUND(
+              CAST(allocated_amount AS NUMERIC), 
+              2
+            ) ELSE 0 END
+          ) AS oss, 
+          SUM(
+            CASE WHEN q2.portfolio_abbrev = 'DES' THEN ROUND(
+              CAST(allocated_amount AS NUMERIC), 
+              2
+            ) ELSE 0 END
+          ) AS des, 
+          SUM(
+            CASE WHEN q2.portfolio_abbrev = 'DMS' THEN ROUND(
+              CAST(allocated_amount AS NUMERIC), 
+              2
+            ) ELSE 0 END
+          ) AS dms, 
+          SUM(
+            CASE WHEN q2.portfolio_abbrev = 'DP' THEN ROUND(
+              CAST(allocated_amount AS NUMERIC), 
+              2
+            ) ELSE 0 END
+          ) AS dp, 
+          SUM(
+            CASE WHEN q2.portfolio_abbrev = 'ANA' THEN ROUND(
+              CAST(allocated_amount AS NUMERIC), 
+              2
+            ) ELSE 0 END
+          ) AS ana, 
+          SUM(
+            CASE WHEN q2.portfolio_abbrev = 'SD' THEN ROUND(
+              CAST(allocated_amount AS NUMERIC), 
+              2
+            ) ELSE 0 END
+          ) AS sd, 
+          SUM(
+            CASE WHEN q2.portfolio_abbrev = 'CE' THEN ROUND(
+              CAST(allocated_amount AS NUMERIC), 
+              2
+            ) ELSE 0 END
+          ) AS ce, 
+          SUM(
+            CASE WHEN q2.portfolio_abbrev = 'EDS' THEN ROUND(
+              CAST(allocated_amount AS NUMERIC), 
+              2
+            ) ELSE 0 END
+          ) AS eds, 
+          SUM(
+            CASE WHEN q2.portfolio_abbrev = 'BCS' THEN ROUND(
+              CAST(allocated_amount AS NUMERIC), 
+              2
+            ) ELSE 0 END
+          ) AS bcs, 
+          SUM(
+            CASE WHEN q2.portfolio_abbrev = 'DIV' THEN ROUND(
+              CAST(allocated_amount AS NUMERIC), 
+              2
+            ) ELSE 0 END
+          ) AS div, 
+          SUM(
+            CASE WHEN q2.portfolio_abbrev = 'GC' THEN ROUND(
+              CAST(allocated_amount AS NUMERIC), 
+              2
+            ) ELSE 0 END
+          ) AS gc, 
+          SUM(q2.resource_count) AS total_resource_count 
+        FROM 
+          q2 
+        GROUP BY 
+          q2.fiscal_year_id, 
+          q2.fy, 
```

Comment:
lol EIGHT SUBQUERIES!

I bet a refactor of the SQL would cut the subquery amounts in half or less